File: swea/D2/1859.백만 장자 프로젝트/1859.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int T;

int main(){
  cin >> T;

  for(int t = 1; t <= T; t++){
    int n;
    cin >> n;

    vector<int> arr(n);
    for(int i = 0; i < n; i++) cin >>arr[i];

    reverse(arr.begin(), arr.end());
    
    int max_p = 0;
    long long ans = 0;
    for(int x : arr){
      if(x > max_p) max_p = x;
      else ans += max_p - x;
    }

    cout << '#' << t << ' ' << ans << endl;
  }
}

File: swea/D2/1859.백만 장자 프로젝트/1859.py
T = int(input())
for tc in range(1, T+1):
    n = int(input())
    arr = list(map(int, input().split()))

    max_p = 0
    ans = 0

    for p in reversed(arr):
        if p > max_p:
            max_p = p
        else:
            ans += max_p - p
    
    print(f'#{tc} {ans}')